gui-tk/wiring.tcl

```diff
@@ -418,8 +418,9 @@ proc textOK {} {
 			minsky.addVariable $name flow
 			minsky.canvas.itemFocus.init $init
             minsky.variableValues.reset
-        } else {  # signed numbers create constant on the canvas. for ticket 145
-			if [regexp "(?:^|\s)(\[+-\]?\[\[:digit:\]\]*\.?\[\[:digit:\]\]+)(?=\$|\s)" $textBuffer] { #https://stackoverflow.com/questions/33520934/regex-with-only-numbers-in-a-string-c          
+# signed numbers create constant on the canvas. for ticket 145. See https://stackoverflow.com/questions/33520934/regex-with-only-numbers-in-a-string-c and https://stackoverflow.com/questions/44771741/regex-any-alphabet-except-e            
+        } else {  
+			if {[regexp "(?:^|\s)(\[+-\]?\[\[:digit:\]\]*\.?\[\[:digit:\]\]+)(?=\$|\s)" $textBuffer] && [regexp "(?:^|\s)(\[^a-df-zA-DF-Z_^\]+$)(?=\$|\s)" $textBuffer]} {          
```

Comment:
This could be more easily achieved by a simpler single regexp. From first principles, I come up with:
"^(\[+-\]?\\d*\\.?\\d+)$"
Comparing this with what you've got, I think the problem is not enough backslashes on the middle ., ie it should be \\.?, not \.?. The problem is that TCL converts \. -> ., which then matches any character, not just the full stop.

Reply:
Oh - I just realised that the parentheses are not necessary either (we just asking for a match, not obtaining subexpressions).

Reply:
I'm trying the combination

 if {[regexp "\[+-\]?\\d*\\.?\\d+" $textBuffer] && [regexp "\[^a-df-zA-DF-Z_^\]+$" $textBuffer]} 

and it works for, e.g., -123e21, but not for a1 or e1 or v1.

However, 

if {[regexp "\[+-\]?\\d*\\.?\\d+" $textBuffer] && [regexp "(?:^|\s)(\[^a-df-zA-DF-Z_^\]+$)(?=\$|\s)" $textBuffer]}

does work. 

I guess that is still an improvement?

Reply:
I was forgetting about the exponent part. How about "^\[+-\]?\\d*\\.?\\d+\[eE\]?\\d*$" ?

PS a tip: Use the Command option in the file menu, and type in the TCL commands directly:
~~~~
gui-tk/minsky>regexp "^(\[+-\]?\\d*\\.?\\d+)$" "1.2"
1
gui-tk/minsky>regexp "^(\[+-\]?\\d*\\.?\\d+)$" "1e2"
0
gui-tk/minsky>regexp "^\[+-\]?\\d*\\.?\\d+[eE]?\d*$" "1e2"
regexp "^\[+-\]?\\d*\\.?\\d+[eE]?\d*$" "1e2"->invalid command name "eE"
0
gui-tk/minsky>regexp "^\[+-\]?\\d*\\.?\\d+[eE]?\\d*$" "1e2"
regexp "^\[+-\]?\\d*\\.?\\d+[eE]?\\d*$" "1e2"->invalid command name "eE"
0
gui-tk/minsky>regexp "^\[+-\]?\\d*\\.?\\d+\[eE\]?\\d*$" "1e2"
1
gui-tk/minsky>regexp "^\[+-\]?\\d*\\.?\\d+\[eE\]?\\d*$" "e2"
0
gui-tk/minsky>regexp "^\[+-\]?\\d*\\.?\\d+\[eE\]?\\d*$" "e_2"
0
gui-tk/minsky>
~~~~

Reply:
Thanks, this was very helpful :)